lib/galaxy/jobs/__init__.py

```diff
@@ -1090,6 +1090,12 @@ def working_directory(self):
                 job, base_dir='job_work', dir_only=True, obj_dir=True)
         return self.__working_directory
 
+    def working_directory_exists(self):
+        job = self.get_job()
```

Comment:
You can move this line after the `if`, to save a database query when returning `False`.

Reply:
Good point!

Reply:
Actually that doesn't work, we need it for `job.object_store_id is None:`

Reply:
Ouch, sorry!

Reply:
but the entire check is wrong, this should just be `return self.app.object_store.exists(job, base_dir='job_work', dir_only=True, obj_dir=True)`, otherwise we return `False` when we shouldn't. Not sure what I was thinking here.

Reply:
That I was trying to understand :)

Reply:
Rebased with the fix, thanks for thinking this through!